agenta-web/src/components/Layout/Layout.tsx

```diff
@@ -120,6 +120,36 @@ const App: React.FC<LayoutProps> = ({children}) => {
         }
     }, [appTheme])
 
+    const computePlaygroundBreadCrumbs = () => {
+        const playground = `/playground`
+        if (router?.pathname?.includes(playground)) {
+            const inclusions = {apps: "apps", [appName]: `apps/${appName}/playground`}
+            const paths = router.asPath
+            const urlPaths = paths?.replace(playground, "").slice(1, paths.length).split("/")
+            if (urlPaths?.length) {
+                return urlPaths.map((u) => {
+                    const decoded = decodeURI(u)
+                    const isLink = inclusions[decoded]
+
+                    if (isLink) {
+                        return {
+                            title: (
+                                <Link href={`/${isLink}`}>
+                                    {renameVariablesCapitalizeAll(decoded)}
+                                </Link>
+                            ),
+                        }
+                    }
+                    return {title: renameVariablesCapitalizeAll(decoded)}
+                })
+            }
+        }
+
+        return [{title: <Link href="/apps">Apps</Link>}, {title: capitalizedAppName}]
+    }
+
+    const breadCrumbItems = computePlaygroundBreadCrumbs()
```

Comment:
please wrap this inside useMemo to avoid unnecessary recomputations:

`const breadCrumbItems = useMemo(computePlaygroundBreadCrumbs, [capitalizedAppName, router.pathname])`

Please include any other dependencies if I've missed